src/Workspaces/Core/Portable/Log/KeyValueLogMessage.cs

```diff
@@ -15,7 +15,7 @@ namespace Microsoft.CodeAnalysis.Internal.Log
     /// <summary>
     /// LogMessage that creates key value map lazily
     /// </summary>
-    internal sealed class KeyValueLogMessage : LogMessage
+    internal sealed class KeyValueLogMessage : LogMessage, IDisposable
```

Comment:
Should this be done for LogMessage, not just this derived type? The concept of Free() exists from there too.

Reply:
As mentioned in the PR description, I didn't want to tackle too much as part of this PR. But, now that someone has indicated this approach makes sense, I'll go ahead and move the other usages of KeyValueLogMessage to this form too.

Reply:
Still considering the right approach here. The Logger.Log* calls take in the LogMessage and Free it, whereas other places that take in a LogMessage (specifically TelemetringLoggin) don't assume ownership.

Perhaps The TelemetryLogging methods that take in a LogMessage should actually assume ownership, in which case, I should get rid of the dispose pattern in this PR as the code freeing the object is no longer the one creating it.

Need to let this stew overnight.

Reply:
@jasonmalinowski -- I reset your vote as I changed the approach here. It's the approach that the Logger class took, and it would be weird for KeyValueLogMessages passed into TelemetryLogging methods to have different semantics (not that I love the approach, but it's what we've got and we should be consistent)